docs/tools/offline/cli.md

```diff
@@ -438,13 +342,190 @@ $ mbed compile -t GCC_ARM -m K64F --profile mbed-os/tools/profiles/debug.json
+
+* Discard local uncommitted changes: *clean* update.
+
+Run `mbed update [branch|revision|tag_name] --clean`
+
+Specifying a branch to `mbed update` will only check out that branch and won't automatically merge or fast-forward to the remote/upstream branch. You can run `mbed update` to merge (fast-forward) your local branch with the latest remote branch. On Git, you can do `git pull`.
+
+<span class="warnings">**Warning**: The `--clean` option tells Mbed CLI to update that program or library and its dependencies and discard all local changes. This action cannot be undone; use with caution.</span>
+
+###### Combine update options
+
+You can combine the options of the Mbed update command for the following scenarios:
+
+* `mbed update --clean --clean-deps --clean-files` - Update the current program or library and its dependencies, remove all local unpublished libraries, discard local uncommitted changes and remove all untracked or ignored files. This wipes every single change that you made in the source tree and restores the stock layout.
+
+* `mbed update --clean --ignore` - Update the current program or library and its dependencies, but ignore any local repositories. Mbed CLI updates whatever it can from the public repositories.
+
+Use these with caution because your uncommitted changes and unpublished libraries cannot be restored.
+
+##### Mbed CLI configuration
+
+You can streamline many options in Mbed CLI with global and local configuration.
+
+The Mbed CLI configuration syntax is:
+
+```
+mbed config [--global] <var> [value] [--unset]
+```
+
+* The **global** configuration (via `--global` option) defines the default behavior of Mbed CLI across programs unless overridden by *local* settings.
+* The **local** configuration (without `--global`) is specific to the Mbed program and allows overriding of global or default Mbed CLI settings.
+* If you do not specify a value, then Mbed CLI prints the value for this setting in this context.
+* The `--unset` option allows you to remove a setting.
+* The `--list` option allows you to list global and local configuration.
+
+Here is a list of configuration settings and their defaults:
+
+ * `target` - defines the default target for `compile`, `test` and `export`; an alias of `mbed target`. Default: none.
+ * `toolchain` - defines the default toolchain for `compile` and `test`; can be set through `mbed toolchain`. Default: none.
+ * `ARM_PATH`, `GCC_ARM_PATH`, `IAR_PATH` - defines the path to Arm Compiler, GCC Arm and IAR Workbench toolchains. Default: none.
+ * `protocol` - defines the default protocol used for importing or cloning of programs and libraries. The possible values are `https`, `http` and `ssh`. Use `ssh` if you have generated and registered SSH keys (Public Key Authentication) with a service such as GitHub, GitLab, Bitbucket and so on. Read more about SSH keys [here](https://help.github.com/articles/generating-an-ssh-key/). Default: `https`.
```

Comment:
I think if you're working with private repos you need to know this, so it's maybe a worthwhile thing to call it out more explicitly